platform/android/dir_access_jandroid.cpp

```diff
@@ -154,7 +154,7 @@ Error DirAccessJAndroid::change_dir(String p_dir){
 
 String DirAccessJAndroid::get_current_dir(){
 
-	return "/"+current_dir;
+	return "res://"+current_dir;
```

Comment:
I tested it more for clear.

```gdscript
func _ready():
	var dir = Directory.new()
	var result = dir.open("res://")
	print(dir.get_current_dir().plus_file("test"))

	var dir = Directory.new()
	var result = dir.open("user://")
	print(dir.get_current_dir().plus_file("test"))

	var dir = Directory.new()
	var result = dir.open("/")
	print(dir.get_current_dir().plus_file("test"))
```
This prints this on desktop and android both. :)
```
res://test
user://test
/test
```